src/main/java/com/redhat/rhjmc/containerjfr/commands/internal/PortScanCommand.java

```diff
@@ -48,18 +51,44 @@ public boolean validate(String[] args) {
 
     @Override
     public void execute(String[] args) throws Exception {
-        scan().forEach(m -> cw.println(String.format("%s -> %s", m.hostname, m.ip)));
+        findCompatibleJvms().forEach(m -> cw.println(String.format("%s -> %s", m.hostname, m.ip)));
     }
 
     @Override
     public Output<?> serializableExecute(String[] args) {
         try {
-            return new ListOutput<>(scan());
+            return new ListOutput<>(findCompatibleJvms());
         } catch (Exception e) {
             return new ExceptionOutput(e);
         }
     }
 
+    private List<IpHostMapping> findCompatibleJvms() throws UnknownHostException, InterruptedException {
+        // Check for environment variables added by Kubernetes that indicate
+        // IP addresses listening on port 9091.
+        List<IpHostMapping> mapping = System.getenv().entrySet().parallelStream()
+                .filter(e -> e.getKey().endsWith(KUBERNETES_ENV_SUFFIX))
+                .map(e -> {
+                    try {
+                        return testHost(InetAddress.getByName(e.getValue()));
+                    } catch (IOException ignored) {
+                        return null;
+                    }
```

Comment:
Could you move the exception handling and `null` return into `testHost` so that the `.map` here can be a one-liner?

Reply:
Sure. These also catch the UnknownHostException thrown by `InetAddress.getBy{Name,Address}`, so I also added `testHostBy{Name,Address}` helpers to catch these as well. I think it looks cleaner now.